The Game/Assets/Scripts/Agent/Player/Cursor_UI_3DSpace.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections;
```

Comment:
you are on the right track but trying but making a complicated function to solve 2 problems

what you need is a helper class

so think of of a public static class that has no monobehaviour

2 functions

get canvas space from mouse position

get world space from mouse position

they need to take in a camera, which the managers calling them should have a reference too

does this make sense?

that way this weird class doesnt need an update function and only the manager does